dartagnan/src/test/java/com/dat3m/dartagnan/exceptions/RuntimeExceptionsTest.java

```diff
@@ -0,0 +1,75 @@
+package com.dat3m.dartagnan.exceptions;
+
+import com.dat3m.dartagnan.expression.Atom;
+import com.dat3m.dartagnan.expression.BConst;
+import com.dat3m.dartagnan.expression.BNonDet;
+import com.dat3m.dartagnan.expression.IExprBin;
+import com.dat3m.dartagnan.expression.INonDet;
+import com.dat3m.dartagnan.expression.INonDetTypes;
+import com.dat3m.dartagnan.expression.IfExpr;
+import com.dat3m.dartagnan.expression.op.COpBin;
+import com.dat3m.dartagnan.expression.op.IOpBin;
+import com.dat3m.dartagnan.parsers.program.utils.ProgramBuilder;
+import com.dat3m.dartagnan.program.Register;
+import com.dat3m.dartagnan.program.event.CondJump;
+import com.dat3m.dartagnan.program.event.Skip;
+import org.junit.Test;
+import org.sosy_lab.java_smt.api.ProverEnvironment;
+import org.sosy_lab.java_smt.api.SolverContext;
+import org.sosy_lab.java_smt.api.SolverContext.ProverOptions;
+
+public class RuntimeExceptionsTest {
```

Comment:
In this whole class I would double check the type of exception you expect. We have the tendency to throw a lot of `RuntimeException` rather than a more specific type of error. I think we should be more concrete about errors and add a lot more custom exception classes for this.

Reply:
I implemented two new types of exceptions for the cases in this test: `TypeException` for problems related to precision (i.e., type) and `BuildException` for problems related to building or validating the program.
But I agree that we might want to implement similar changes to the overall code.
I will keep this in mind while I add new unit tests to this PR.

Reply:
I would make the exception names more informative and precise: `TypeMismatchException` (maybe even `ExprTypeMismatchException`) and `InvalidProgramException` (or `MalformedProgramException)`.
The name `BuildException` only makes sense because we have a class `ProgramBuilder` but if you renamed that class, the exception name would become odd. This indicates that the exception name is not informative about the type of error that occurred but only about where the error occurred, which I think is wrong design.